packages/ilios-common/addon/components/course/rollover.hbs

```diff
@@ -46,6 +40,7 @@
             id="year-{{templateId}}"
             data-test-year
             {{on "change" this.setSelectedYear}}
+            {{get-element this.loadValidYear}}
```

Comment:
This usage doesn't track for me though. There isn't anything here that needs access to the DOM. This data should be loaded in a TrackedAsyncData and processed in a getter.

Reply:
The reason for adding that there is that the `Course::RolloverDatePicker` is waiting for the **Year** `<select>` element to load before it gets the `selectedYear` + 1 value. If I don't have that there, then the **Year** `<select>` loads, say, `2025`, and the date picker remains on `11/26/2024` instead of rolling over to `11/25/2025` like it currently does on demo.

Reply:
How about wrapping everything in an `if` so it only tries to render once the data needed is resolved?

Reply:
Is not the whole thing already wrapped in an `if`?

```
{{#if this.allCoursesData.isResolved}}
  <select
    id="year-{{templateId}}"
    data-test-year
    {{on "change" this.setSelectedYear}}
    {{get-element this.loadValidYear}}
  >
    {{#each this.years as |year|}}
      ...
    {{/each}}
  </select>
{{else}}
  <LoadingSpinner />
{{/if}}
```